File: Java/ironsource/src/main/java/vdo/ai/sample/java/ironsource/application/MyApplication.java
package vdo.ai.sample.java.ironsource.application;

import android.app.Application;
import android.content.Context;

import androidx.annotation.Nullable;

import vdo.ai.android.core.VdoAppOpenAd;
import vdo.ai.android.core.listeners.VdoAppOpenListener;
import vdo.ai.android.core.models.VdoAdError;
import vdo.ai.android.ironsource.VdoIronSourceAppOpen;
import vdo.ai.sample.java.ironsource.BuildConfig;


public class MyApplication extends Application {

    public VdoIronSourceAppOpen appOpenAd;
    public static String IRON_SOURCE_APP_KEY ="85460dcd"; // sample app key

    @Override
    public void onCreate() {
        super.onCreate();


        appOpenAd = new VdoIronSourceAppOpen.VdoISAppOpenBuilder()
                .withContext(this)
                .setEnvironment(BuildConfig.BUILD_TYPE)
                .setTagName("in-app-sample-ap-Z1")
                .setIronSourceParams(IRON_SOURCE_APP_KEY, "DefaultOfferWall")
                .setListener(new VdoAppOpenListener(){

                    @Override
                    public void onAdFailedToShowFullScreenContent(@Nullable VdoAdError vdoAdError) {

                    }

                    @Override
                    public void onAdFailedToLoad(@Nullable VdoAdError vdoAdError) {

                    }

                    @Override
                    public void onAdLoaded() {

                    }

                    @Override
                    public void onAdDismissedFullScreenContent() {

                    }

                    @Override
                    public void onAdShowedFullScreenContent() {

                    }
                }).build();
    }

    @Override
    protected void attachBaseContext(Context base) {
        super.attachBaseContext(base);
        // if using MultiDexApplication not required Z1MediaManager.initMultiDex function
    }
}

File: Kotlin/example-ironsource/src/main/java/vdo/ai/android/example/ironsource/application/MyApplication.kt
package vdo.ai.android.example.ironsource.application

import android.app.Application
import android.content.Context
import android.util.Log

import vdo.ai.android.core.listeners.VdoAppOpenListener
import vdo.ai.android.core.models.VdoAdError
import vdo.ai.android.example.ironsource.BuildConfig
import vdo.ai.android.example.ironsource.ironSourceOfferApiKey
import vdo.ai.android.ironsource.VdoIronSourceAppOpen

/**
 *  created by Harsh Shrivastava at 6th Oct 2023
 *
 **/
class MyApplication : Application() {

    var appOpenAd: VdoIronSourceAppOpen?= null
    var TAG= MyApplication :: class.java.simpleName

    override fun onCreate() {
        super.onCreate()

        appOpenAd = VdoIronSourceAppOpen.VdoISAppOpenBuilder()
            .withContext(this@MyApplication)
            .setEnvironment(BuildConfig.BUILD_TYPE)
            .setTagName("in-app-sample-ap-Z1")
            .setIronSourceParams(ironSourceOfferApiKey, "DefaultOfferWall")
            .setListener(object:VdoAppOpenListener{
                override fun onAdLoaded() {
                }

                override fun onAdFailedToLoad(adError: VdoAdError?) {
                }

                override fun onAdDismissedFullScreenContent() {
                }

                override fun onAdShowedFullScreenContent() {
                }

                override fun onAdFailedToShowFullScreenContent(adError: VdoAdError?) {
                }

            }).build()

    }

    override fun attachBaseContext(context: Context) {
        super.attachBaseContext(context)
//        Z1MediaManager.initMultiDex(this)
    }
}

